front/src/applications/operationalStudies/components/MacroEditor/utils.ts

```diff
@@ -238,3 +240,21 @@ export const getTrainrunFrequencyFromTimetableItem = (
   }
   return trainrunFrequency;
 };
+
+export const getNetzgrafikColors = () =>
+  Object.entries(NETZGRAFIK_COLOR_PALETTE).map(([colorRef, baseColor], index) => {
+    const key = baseColor;
+
+    return {
+      id: index,
+      colorRef,
+      color: CATEGORY_COLOR_VARIANTS[key][50],
+      colorFocus: CATEGORY_COLOR_VARIANTS[key][70],
+      colorMuted: CATEGORY_COLOR_VARIANTS[key][10],
+      colorRelated: CATEGORY_COLOR_VARIANTS[key][50],
+      colorDarkMode: CATEGORY_COLOR_VARIANTS[key][50],
+      colorDarkModeFocus: CATEGORY_COLOR_VARIANTS[key][70],
+      colorDarkModeMuted: CATEGORY_COLOR_VARIANTS[key][10],
+      colorDarkModeRelated: CATEGORY_COLOR_VARIANTS[key][50],
```

Comment:
it's a lot of repetion, maybe you could have a const `CATEGORY_COLOR_VARIANTS[key]`